pinot-api/src/main/java/org/apache/pinot/client/PinotClientTransportFactory.java

```diff
@@ -18,9 +18,11 @@
  */
 package org.apache.pinot.client;
 
+import java.util.Map;
+
 /**
  * Factory for client transports.
  */
 interface PinotClientTransportFactory {
-  PinotClientTransport buildTransport();
+  PinotClientTransport buildTransport(Map<String, String> headers);
```

Comment:
This is an interface change and can potentially impact other users of this module. Please follow Pinot guidelines for public interface changes.

Reply:
+1

It looks like this can be extended by others and this signature change will break compatibility. This interface should have been marked as public:stable. We recently add annotations for a few foundation classes, we'll need to do the same here.

@jitendratheta, if you can review the guidelines here: https://github.com/apache/incubator-pinot/blob/master/pinot-common/src/main/java/org/apache/pinot/annotations that will be great. Javadocs for InterfaceStability.java provide more specific info. As noted there, to keep compatibility, the best option is to provide a default implementation for the new method while keeping or marking the old one as deprecated. 

We'll look into making one more pass at updating interface annotations, but for now, it looks like this class should have Audience.Public and Stability.Stable annotations. Could you add the same?

@mcvsubbu, @Jackie-Jiang - FYI. Please chime in as appropriate.

Reply:
I had the same impression. But ConnectionFactory is the public interface and he has maintained backward compatibility by adding a new method.

Reply:
@sunithabeeram as per your recommendation I made this interface as public:stable.

This interface is not exposed as a public interface
  1. should we make it `public interface PinotClientTransportFactory`?
  2. If we do so, PinotClientTransportFactory is not accepted as an argument by any class. ConnectionFactory class is using it by implicit object creation inside the class. then does it make sense to have it public? or should we also make `PinotClientTransport` as public and one of the argument to be passed to function `ConnectionFactory.fromHostList` and let the caller decide the implementation of it?

Reply:
Or else I was thinking why not to keep the annotation as Private for class `PinotClientTransportFactory` as it is not exposed outside as public?

Reply:
I don't see anyone extending PinotClientTransportFactory outside of Pinot code. I am fine with making it private

Reply:
@sunithabeeram @mayankshriv Can this thread be resolved if you are ok with the changes?

Reply:
The code structure seems a bit weird to me. If ConnectionFactory is the one that can be extended, then shouldn't that actually be an interface and made Public.Stable? Its odd that we have a specific implementation there and expect clean extensibility.

Given the lack of clarity, it would be great if the javadocs for ConnectionFactory and PinotClientConnectionFactory be updated to expand on how they should be used - ie, what should change when new transports are considered/added. @kishoreg, hope you can provide appropriate input here for Jitendra.

Reply:
@sunithabeeram, Had a discussion with @kishoreg and added discussed changes.

Reply:
@kishoreg it looks like the current ConnectionFactory is tied to JsonHttp transport. How would things work if Https has to be supported? It appears to me that the transport factory itself needs to be injected to ConnectionFactory class in that case. 

The class has been marked as evolving. Is it because you see new methods getting added in the future or do you have other cleanup/changes in mind?